src/Compiler/SyntaxTree/SyntaxTreeOps.fs

```diff
@@ -351,6 +351,15 @@ let rec SimplePatsOfPat synArgNameGenerator p =
 
     | SynPat.Const (SynConst.Unit, m) -> SynSimplePats.SimplePats([], [], m), None
 
+    | SynPat.Paren (SynPat.As (lhsPat = SynPat.Tuple (elementPats = pats; commaRanges = commas); rhsPat = rhsPat), m) ->
```

Comment:
I'd maybe suggest trying to add a `SynPat.Paren` branch, similar to the `SynPat.FromParseError` one above, to unwind parens from other pattern kinds. Then it'll be easier to process compound patterns like tuples and `as`-patterns below it. And it will also cover cases where nested patterns are used, e.g. `((a as b))`, `((a, b as c))`.

Reply:
Agreed. There are a lot of patterns that we could process TBH. To make it safe I will do follow-up PRs :)